src/token/api/src/canister.rs

```diff
@@ -133,28 +137,23 @@ pub trait TokenCanisterAPI: Canister + Sized {
     }
 
     #[query(trait = true)]
-    fn getHolders(&self, start: usize, limit: usize) -> Vec<(Principal, Tokens128)> {
+    fn getHolders(&self, start: usize, limit: usize) -> Vec<(Account, Tokens128)> {
         self.state().borrow().balances.get_holders(start, limit)
     }
 
     #[query(trait = true)]
-    fn getAllowanceSize(&self) -> usize {
-        self.state().borrow().allowance_size()
-    }
-
-    #[query(trait = true)]
-    fn getUserApprovals(&self, who: Principal) -> Vec<(Principal, Tokens128)> {
-        self.state().borrow().user_approvals(who)
-    }
-
-    #[query(trait = true)]
-    fn balanceOf(&self, holder: Principal) -> Tokens128 {
-        self.state().borrow().balances.balance_of(&holder)
+    fn icrc1_balance_of(
```

Comment:
To match the spec, these arguments needs to wrapped in a struct, see this:

 icrc1_balance_of : (record { of: principal; subaccount: opt SubAccount; }) -> (nat) query;

Reply:
Ok